### PR TITLE
Paginated Users List

### LINKED ISSUE #234: Paginated user list
- [x] Add a function to derive data for rendering pagination widget
- [x] Add pagination widget
- [x] Switch to use page number based pagination in user list

### PR DESCRIPTION
**Pull request by louischan-oursky**
fix #234

### CODE REVIEW

File: portal/src/util/pagination.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  currentOffset: number;
+  offsets: number[];
+  firstPageButtonEnabled: boolean;
+  prevPageButtonEnabled: boolean;
+  nextPageButtonEnabled: boolean;
+  lastPageButtonEnabled: boolean;
+  maxOffset?: number;
+}
+
+const PAGE_OFFSET = 3;
```
Comment (kiootic) [inline portal/src/util/pagination.ts:17]: I think we should add an explanation on what it is.
If I understood correctly, it is the number of pages before/after the current page shown in the page list?

**Review by kiootic — COMMENTED** [review]
I tested with a 10-item first page and 1-item second page, and the pagination widget jumps around when switching page. I think it should be fixed at bottom of the page?

File: portal/src/util/pagination.ts (hunk 1 above)
Comment (louischan-oursky) [reply portal/src/util/pagination.ts:17]: Yes.

**Comment by louischan-oursky** [comment]
Added 2 commits.

**Review by kiootic — APPROVED** [review]

**Pull request merged**